src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -148,14 +179,14 @@ await EventGridPublisherBuilder
                 .PublishManyAsync(events);
 
             // Assert
-            Assert.All(
-                events,
-                e =>
-                {
-                    TracePublishedEvent(e.Id, events);
-                    string receivedEvent = _serviceBusEventConsumerHost.GetReceivedEvent(e.Id, 5);
-                    AssertReceivedEvent(e.Id, e.EventType, e.Subject, e.Data.LicensePlate, receivedEvent);
-                });
+            Assert.All(events, AssertReceivedNewRegisteredCarEventWithRetryCount);
+        }
+
+        private void AssertReceivedNewRegisteredCarEventWithRetryCount(NewCarRegistered @event)
```

Comment:
The extracted methods are a lot of duplication where the only difference is the timeout for one test, I think.

Is there a reason why we are not re-using them?

Reply:
Well, I try not to re-use so much in the tests like you said to not avoid beginner devs to loose sight.

Reply:
I think it actually more the opposite - It would be a lot easier if it's just re-used so that the tests are a lot cleaner.

Reply:
I aggree, that was why I always intensively use propery-based testing data-driven tests, but I was under the impression that this and even theories are 'basic' enough; so is reusing of arrange/act/assert functionality.

Reply:
If you assert same things it's ok to refactor into one method so feel free to refactor the new methods into one